src/test/java/uk/gov/hmcts/reform/unspec/handler/RespondExtensionCallbackHandlerTest.java

```diff
@@ -0,0 +1,163 @@
+package uk.gov.hmcts.reform.unspec.handler;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.SubmittedCallbackResponse;
+import uk.gov.hmcts.reform.unspec.callback.CallbackParams;
+import uk.gov.hmcts.reform.unspec.callback.CallbackType;
+import uk.gov.hmcts.reform.unspec.validation.RequestExtensionValidator;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.google.common.collect.ImmutableMap.of;
+import static java.lang.String.format;
+import static java.time.LocalDate.now;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.unspec.helpers.DateFormatHelper.DATE;
+import static uk.gov.hmcts.reform.unspec.helpers.DateFormatHelper.formatLocalDateTime;
+
+@SpringBootTest(classes = {
+    RespondExtensionCallbackHandler.class,
+    RequestExtensionValidator.class,
+    JacksonAutoConfiguration.class
+})
+class RespondExtensionCallbackHandlerTest extends BaseCallbackHandlerTest {
+
+    public static final String RESPONSE_DEADLINE = "responseDeadline";
+    public static final String COUNTER_DATE = "respondentSolicitor1claimResponseExtensionCounterDate";
+
+    @Autowired
+    private RespondExtensionCallbackHandler handler;
+
+    @Nested
+    class AboutToStartCallback {
+
+        public static final String EXTENSION_REASON = "respondentSolicitor1claimResponseExtensionReason";
+
+        @Test
+        void shouldAddNoReasonGiven_WhenNoReasonGivenForExtensionRequest() {
+            CallbackParams params = callbackParamsOf(new HashMap<>(), CallbackType.ABOUT_TO_START);
+
+            AboutToStartOrSubmitCallbackResponse response =
+                (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+
+            assertThat(response.getData().get(EXTENSION_REASON)).isEqualTo("No reason given");
```

Comment:
Minor: can be simplified to use the corresponding dedicated assertion. i.e. 
```suggestion
            assertThat(response.getData()).containsEntry(EXTENSION_REASON, "No reason given");
```